README.md

```diff
@@ -195,52 +195,78 @@ Install custom reporters as a development dependency and you can use the `--repo
 nyc report --reporter=<custom-reporter-name>
 ```
 
-## Excluding files
+## Selecting files for coverage
 
-You can tell nyc to exclude specific files and directories by adding
-an `nyc.exclude` array to your `package.json`. Each element of
-the array is a glob pattern indicating which paths should be omitted.
+By default, nyc only collects coverage for source files that are visited
+during a test.  It does this by watching for files that are `require()`'d 
+during the test.  Only source files that are visited during a test will appear 
+in the coverage report and contribute to coverage statistics.  
+
+Nyc will cover all files if the `--all` flag is set.  In this case all files will
+appear in the coverage report and contribute to coverage statistics.
+
+Nyc will only cover files that are located under `cwd`, and then only `*.js` files
+or files with extensions listed in in the `nyc.extension` array.
+
+You can reduce the set of covered files by adding `include` and `exclude` 
+filter arrays to your config.  These allow you to shape the set of covered files 
+by specifying glob patterns that can filter files from  the covered set.  The `exclude` 
+array may also use exclude negated glob patterns, these are specified with 
+a `!` prefix, and can restore sub-paths of excluded paths.
 
 Globs are matched using [minimatch](https://www.npmjs.com/package/minimatch).
 
-For example, the following config will exclude any files with the extension `.spec.js`,
-and anything in the `build` directory:
+We use the following process to remove files from consideration,  
+ * First, limit the set of covered files to those files in paths listed in the 
+   `include` array.  
+ * Then, remove any files that are found in the `exclude` array.
+ * Finally, restore any exclude negated files that have been excluded in the 
+   second step
 
-```json
-{
-  "nyc": {
-    "exclude": [
-      "**/*.spec.js",
-      "build"
-    ]
-  }
-}
-```
-> Note: Since version 9.0 files under `node_modules/` are excluded by default.
-  add the exclude rule `!**/node_modules/` to stop this.
 
-> Note: exclude defaults to `['coverage/**', 'test/**', 'test{,-*}.js', '**/*.test.js', '**/__tests__/**', '**/node_modules/**']`,
+##### Using include and exclude arrays
```

Comment:
Shouldn't this just be an h3 to be consistent?

Reply:
I wanted it to seem like a very minor heading in the scope of the whole section.  But it can change to h3